com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -104,29 +116,166 @@ public int AddEntry(ulong networkObjectId, int behaviourIndex, int variableIndex
             return pos;
         }
 
+        /// <summary>
+        /// Write an Entry to send
+        /// Must match ReadEntry
+        /// </summary>
+        /// <param name="writer">The writer to write the entry to</param>
+        internal void WriteEntry(NetworkWriter writer, in Entry entry)
+        {
+            //todo: major refactor.
+            // use blittable types and copy variable in memory locally
+            // only serialize when put on the wire for network transfer
+            writer.WriteUInt64(entry.Key.NetworkObjectId);
+            writer.WriteUInt16(entry.Key.BehaviourIndex);
+            writer.WriteUInt16(entry.Key.VariableIndex);
+            writer.WriteUInt16(entry.Key.TickWritten);
+            writer.WriteUInt16(entry.Position);
+            writer.WriteUInt16(entry.Length);
+        }
+
+        /// <summary>
+        /// Read a received Entry
+        /// Must match WriteEntry
+        /// </summary>
+        /// <param name="reader">The readed to read the entry from</param>
+        internal Entry ReadEntry(NetworkReader reader)
```

Comment:
Can be 'in' modified, no?

Reply:
Why? `NetworkReader` is a reference type. I thought `in` was meant to improve the performance by not copying value types. If it was an `in` parameter, it would only prevent assigning `reader` a new NetworkReader. Since the code doesn't do it, since it wouldn't affect the caller if it did, and since there's no performance improvements in doing it, I fail to see the point.